packages/hardhat-foundry/DESIGN.md

```diff
@@ -0,0 +1,40 @@
+# Design
```

Comment:
This explanation, but with less technical details, should be in the readme and the guide

Reply:
Otherwise it feels like adding an unspecified blob of functionality